### PR TITLE
be consistent in the use of Logs vs Log module

### PR DESCRIPTION
**Pull request by palainp**
Dear developpers, the goal for this PR is to print a more specific message in the logs when the library (e.g. so far it prints `DBG [application]` sometimes and `DBG [more-specific]` othertimes, now it always provide the source module in the logs).

I just kept `app/*` with `Logs.*` as it will be displayed as `[application]` and it seems fine.

### CODE REVIEW

File: resolver/dns_resolver.ml [hunk 1]
```diff
@@ -8,6 +8,9 @@ type key = [ `raw ] Domain_name.t * Packet.Question.qtype
 
 let pp_key = Dns_resolver_cache.pp_question
 
+let src = Logs.Src.create "dns-resolver" ~doc:"DNS resolver"
```
Comment (hannesm) [inline resolver/dns_resolver.ml:11]:
```suggestion
let src = Logs.Src.create "dns_resolver" ~doc:"DNS resolver"
```

let's stay consistent across this repository.

File: resolver/dns_resolver_cache.ml [hunk 2]
```diff
@@ -4,6 +4,9 @@ open Dns
 
 module N = Domain_name.Set
 
+let src = Logs.Src.create "dns-resolver-cache" ~doc:"DNS resolver cache"
```
Comment (hannesm) [inline resolver/dns_resolver_cache.ml:7]:
```suggestion
let src = Logs.Src.create "dns_resolver_cache" ~doc:"DNS resolver cache"
```

File: resolver/dns_resolver_utils.ml [hunk 3]
```diff
@@ -3,6 +3,9 @@
 open Dns
 open Dns_resolver_cache
 
+let src = Logs.Src.create "dns-resolver-util" ~doc:"DNS resolver util"
```
Comment (hannesm) [inline resolver/dns_resolver_utils.ml:6]:
```suggestion
let src = Logs.Src.create "dns_resolver_util" ~doc:"DNS resolver util"
```

**Comment by hannesm** [comment]
thanks a lot

File: resolver/dns_resolver.ml (hunk 1 above)
Comment (palainp) [reply resolver/dns_resolver.ml]: Yes of course, sorry for that.

**Pull request merged**